js/src/run_trees.ts

```diff
@@ -41,16 +43,52 @@ export interface RunTreeConfig {
   client?: Client;
 }
 
+export interface RunnableConfigLike {
+  /**
+   * Tags for this call and any sub-calls (eg. a Chain calling an LLM).
+   * You can use these to filter calls.
+   */
+  tags?: string[];
+
+  /**
+   * Metadata for this call and any sub-calls (eg. a Chain calling an LLM).
+   * Keys should be strings, values should be JSON-serializable.
+   */
+  metadata?: Record<string, unknown>;
+
+  /**
+   * Callbacks for this call and any sub-calls (eg. a Chain calling an LLM).
+   * Tags are passed to all callbacks, metadata is passed to handle*Start callbacks.
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  callbacks?: any;
+}
+
+interface CallbackManagerLike {
+  handlers: TracerLike[];
+  getParentRunId(): string | undefined;
```

Comment:
Would make this an optional param:

`getParentRunId?: () => string | undefined`

To avoid compatibility issues